packages/graphql/lib/src/core/query_options.dart

```diff
@@ -67,12 +69,17 @@ class Policies {
 /// Base options.
 class BaseOptions extends RawOperationData {
   BaseOptions({
-    @required String document,
+    @Deprecated('Use documentNode instead') String document,
```

Comment:
Good to first deprecated support for `document`. Maybe rephrase the message to:

```
'The "document" option has been deprecated, use "documentNode" instead'
```

Reply:
OK, added a more verbose deprecation message